internal/clients/clients.go

```diff
@@ -117,6 +116,7 @@ func Anonymize(cfg *rest.Config) *rest.Config {
 // type the client is asked to get or list. Clients (and their caches) expire
 // and are garbage collected if they are unused for five minutes.
 type Cache struct {
+	ctx    context.Context
```

Comment:
instead of creating an adhoc context when initializing a client, create a context that will be valid for the lifetime of `Cache`